notion-server/src/ethereum/mod.rs

```diff
@@ -279,6 +311,39 @@ pub mod args {
 
     // Return values AND/OR sub-components of incoming argument values
 
+    #[derive(Debug, Default)]
+    pub struct BeaconBlock {
+        pub cross_links: HashMap<u32, CrossLink>,
+    }
+    impl From<&super::BeaconBlock> for BeaconBlock {
+        fn from(beacon_block: &super::BeaconBlock) -> Self {
+            let mut cross_links: HashMap<u32, CrossLink> = HashMap::new();
+            for (k, v) in &beacon_block.cross_links {
+                let super::ShardChainIndex(shard_chain_index) = k;
+                let cross_link: CrossLink = CrossLink::from(v);
+                cross_links.insert(*shard_chain_index, cross_link);
+            }
+            Self { cross_links }
+        }
+    }
+
+    #[derive(Default, Debug, Clone)]
+    pub struct CrossLink {
+        pub execution_environment_states: HashMap<u32, Vec<u8>>,
```

Comment:
In [other places](https://github.com/quilt/simulation/blob/15c6229746160835153f2a40bb386e3f6373c8d6/notion-server/src/ethereum/mod.rs#L500) the execution environment state is a `[u8; 32]`. I'm curious why a `Vec<u8>` here?

Reply:
Also why is this a hashmap vs a fixed length vector or array? Zero-indexed shard number - def shouldn't be a hashmap. https://github.com/sigp/lighthouse/blob/master/eth2/types/src/beacon_state.rs

Reply:
Same with cross-links - shard numbers are just indexed - no need for the heaviness of a hashmap.

Reply:
@villanuevawill I think at the time we started using the hashmap pattern we weren't certain of whether the # of shards was set in stone, and hashmap was an easy way to simulate it.  Definitely not attached to the HashMap approach, but easy to change later also.  I think what I'd like to do is land this diff then have 1 diff next that converts from HashMap to array or vector.

@SamWilsn good catch! I'll update that, my mistake.